TheMovieDb/Network/APIError.swift

```diff
@@ -0,0 +1,27 @@
+//
+//  APIError.swift
+//  TheMovieDb
+//
+//  Created by Rob Cruz on 18/11/21.
+//
+
+import Foundation
+
+// MARK: - Error Handling
+
+enum APIError: Error {
+    case requestFailed
+    case jsonConversionFailure
+    case invalidData
+    case responseUnsuccessful
+    case jsonParsingFailure
+    var localizedDescription: String {
+        switch self {
+        case .requestFailed: return Constants.Errors.requestFailed
+        case .invalidData: return Constants.Errors.InvalidData
+        case .responseUnsuccessful: return Constants.Errors.responseUnsuccessful
+        case .jsonParsingFailure: return Constants.Errors.jsonParsingFailure
+        case .jsonConversionFailure: return Constants.Errors.jsonConversionFailure
+        }
+    }
+}
```

Comment:
`APIError` use is to be throw on `Error`, and provide a `String` description. I would recommend to conform to those 2 protocols. `enum APIError: String, Error`. That way the error "reference" is both throwable and describable.

[Spaghetti code like spaghetti, is conform but not sustainable.](https://betterprogramming.pub/10-tricks-to-avoid-spaghetti-code-in-ios-development-3f5a0ab2f46f)